# RNAIntels
Machine learning-based tool for discovering coding and non coding small RNAs in human RNA-seq data.
## Description
Following receiving raw sequence readings from an organism, next is the need to classify them into their functional units–coding or non-coding sequences. Coding RNA sequences are translated into protein, while non-coding RNAs, e.g., small non-coding RNA such as miRNA or siRNA, are not translated into proteins but are essential for many significant biological processes, such as gene regulation and gene expressions. Although several research efforts are geared towards developing computational tools for functional prediction of RNA-seq data, identifying coding and non-coding RNAs from raw RNA-seq data remains a challenging task. Here we present a software tool using machine-learning approaches to classify short, human RNA sequences (in the range of 28 – 51 nucleotides) into coding and non-coding RNAs. In contrast to many existing approaches, RNAIntels neither requires raw RNA-Seq sequences (FASTQ) nor aligning the sequences to a human reference genome, thus reducing complexities issues that made other tools less attractive to users.
## Requirements
#### Hardware Requirements
RNAIntels requires a  functional computer system with a minimum of 4g RAM for efficient encoding of the RNA-seq data


#### OS Requirements

## Dependancies

## Data
## Running the code
